### DISCUSSION TITLE
TensoMechanics variables definition

### DISCUSSION THREAD

**Discussion by MonikaCarvajal** [idaholab/moose#16086] [Q&A Modules: Solid mechanics] [answered]
Hi all,

I have little to non experience on C++ and I'm learning it on the fly while I use MOOSE.

I have a material distributed in space that I would like to operate it with a simple function in the _Material_ system. What type of variable do I have to define in the head file to store the result of this operation?

More precisely, the head file looks like this
protected:

virtual void computeQpElasticityTensor() override;

  /// Space distributed material
  const VariableValue & _space_distribuited
  /// Storage the operation
  **Variable type** & _operation_result;
};

I hope this makes sense,
Monika

**Comment by aeslaughter** [comment, marked as answer]
Can you provide more details regarding what you are trying to compute. You code indicates that you are using the tensor mechanics module and working on computing an elasticity tensor. Does the value you need to compute contribute to this tensor or are you computing something else?

**Reply by MonikaCarvajal** [reply]
Hi @aeslaughter I solved this issue. It was quite simple hehe. Thanks anyways :)
